### PR TITLE
[ACS-7462] cleanup deprecated api for DataTable and DocumentList

### PR DESCRIPTION
**Pull request by DenysVuika**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [x] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

- deprecate unused and untested gallery mode for document list and datatable (ACS-7462)
- remove unused interface (ACS-7460)
- remove unused `DataTable.iconAltTextKey` method
- remove unused `DataTable.getSortingKey` method (part of gallery UI)
- update documentation for DataTable - bug fixes, typo fixes, example fixes

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: docs/content-services/components/document-list.component.md [hunk 1]
```diff
@@ -90,7 +90,6 @@ Displays the documents from a repository.
 | sorting | `string[] \| `[`DataSorting`](../../../lib/core/src/lib/datatable/data/data-sorting.model.ts) | ['name', 'asc'] | Defines default sorting. The format is an array of 2 strings `[key, direction]` i.e. `['name', 'desc']` or `['name', 'asc']`. Set this value only if you want to override the default sorting detected by the component based on columns. |
 | sortingMode | `"server" \| "client"` | "server" | Defines sorting mode. Can be either `client` (items in the list are sorted client-side) or `server` (the ordering supplied by the server is used without further client-side sorting). Note that the `server` option _does not_ request the server to sort the data before delivering it. |
 | stickyHeader | `boolean` | false | Toggles the sticky header mode. |
-| thumbnails | `boolean` | false | Show document thumbnails rather than icons |
```
Comment (nikita-web-ua) [inline docs/content-services/components/document-list.component.md:93]: I suppose  `display` on line 69 should also be removed

Comment (DenysVuika) [reply docs/content-services/components/document-list.component.md:93]: yes, it was display to remove , not thumbnails

**Review by MichalKinas — APPROVED** [review]
Just one question, should we add a comment somewhere to indicate that the gallery mode won't be available anymore since next version?

**Comment by ath-atolcd** [comment]
> Just one question, should we add a comment somewhere to indicate that the gallery mode won't be available anymore since next version?

Hello,

We understand that you have deprecated and removed the gallery mode, as it was not being used.

However, we use it as a display option in our applications.

Do you think gallery mode could be reintroduced/revised in a future version ?

ty

**Pull request merged**